src/App.tsx

```diff
@@ -15,48 +16,111 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortField {
+  Default = 'none',
+  Name = 'name',
+  Length = 'length',
+}
+
+interface FilterParaments {
+  sortField: string;
+  isReversed: boolean;
+}
+
+function getPrepareGoods(
+  goods: string[],
+  { sortField, isReversed }: FilterParaments,
+): string[] {
+  let prepareGoods = [...goods];
+
+  if (sortField !== SortField.Default) {
+    prepareGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortField.Name:
+          return good1.localeCompare(good2);
+
+        case SortField.Length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    prepareGoods = prepareGoods.reverse();
+  }
+
+  return prepareGoods;
+}
+
+const sortButtons = [
+  { name: 'Sort alphabetically', class: 'is-info', change: SortField.Name },
+  { name: 'Sort by length', class: 'is-success', change: SortField.Length },
+];
+
+type Callback = () => void;
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortField.Default);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleGoods = getPrepareGoods(
+    goodsFromServer,
+    { sortField, isReversed },
+  );
+
+  const onGoodClick = (sortChange: SortField) => () => {
+    setSortField(sortChange);
+  };
+
+  const onReverseClick: Callback = () => {
```

Comment:
```suggestion
  const onReverseClick: = () => {
```

this is redundant here